packages/base/src/formbuilder/objectform/geojsonsource.ts

```diff
@@ -1,18 +1,23 @@
-import { IDict } from '@jupytergis/schema';
+import { IDict, SourceType } from '@jupytergis/schema';
 import { Ajv, ValidateFunction } from 'ajv';
 import * as geojson from '@jupytergis/schema/src/schema/geojson.json';
 
 import { IBaseFormProps } from './baseform';
 import { PathBasedSourcePropertiesForm } from './pathbasedsource';
 import { loadFile } from '../../tools';
 
+export interface IGeoJSONSourceFormProps extends IBaseFormProps {
+  sourceType: SourceType;
+}
```

Comment:
We should probably have a more common `ISourceFormProps` (introduce a new `sourceform.ts` file, similar to `layerform.ts`) that all source forms use. 

```suggestion
export interface ISourceFormProps extends IBaseFormProps {
  sourceType: SourceType;
}
```